pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslMicroBatchInputPartition.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.spark;
+
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.cloudpubsub.FlowControlSettings;
+import com.google.cloud.pubsublite.internal.BlockingPullSubscriber;
+import com.google.cloud.pubsublite.internal.BlockingPullSubscriberImpl;
+import com.google.cloud.pubsublite.internal.CheckedApiException;
+import com.google.cloud.pubsublite.internal.wire.PubsubContext;
+import com.google.cloud.pubsublite.internal.wire.SubscriberBuilder;
+import com.google.cloud.pubsublite.proto.Cursor;
+import com.google.cloud.pubsublite.proto.SeekRequest;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.sources.v2.reader.InputPartition;
+import org.apache.spark.sql.sources.v2.reader.InputPartitionReader;
+
+public class PslMicroBatchInputPartition implements InputPartition<InternalRow> {
+
+  private final SparkPartitionOffset startOffset;
+  private final SparkPartitionOffset endOffset;
+  private final SubscriptionPath subscriptionPath;
+  private final FlowControlSettings flowControlSettings;
+
+  public PslMicroBatchInputPartition(
+      SubscriptionPath subscriptionPath,
+      FlowControlSettings flowControlSettings,
+      SparkPartitionOffset startOffset,
+      SparkPartitionOffset endOffset) {
+    this.startOffset = startOffset;
+    this.endOffset = endOffset;
+    this.subscriptionPath = subscriptionPath;
+    this.flowControlSettings = flowControlSettings;
+  }
+
+  @Override
+  public InputPartitionReader createPartitionReader() {
+    PslPartitionOffset pslPartitionOffset = PslSparkUtils.toPslPartitionOffset(startOffset);
+
+    BlockingPullSubscriber subscriber;
+    try {
+      subscriber =
+          new BlockingPullSubscriberImpl(
+              // TODO(jiangmichael): Pass credentials settings here.
+              (consumer) ->
```

Comment:
nit. I'd push this up the stack a bit. Pass in a SerializableSupplier<BlockingPullSubscriber> instead, then you don't need to think about credentials or the start offset here at all.